### ISSUE TITLE
when using multiple services traefik.port=<int> is still required to be present

### ISSUE THREAD

**Issue by travisghansen** [containous/traefik#1840]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do? 

Tried to launch a swarm service with multiple traefik services

### What did you expect to see?

My multiples traefik services registered

### What did you see instead?

Nothing

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
1.3.2
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```
    --docker \
    --docker.swarmmode=true \
    --docker.exposedbydefault=false \
    --docker.watch=true \
    --docker.domain=example.com \
    --web
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

```
(paste your output here)
```

**Comment by ldez** [comment]
Thanks for your interest in Traefik !

Every question in the issues means features less 😉.

The issue tracker is for reporting bugs and feature requests only.
For end-user related support questions, refer to one of the following:

- [Stack Overflow](https://stackoverflow.com/questions/tagged/traefik) using the "traefik" tag
- the [Traefik community Slack channel](https://traefik.herokuapp.com)

**Comment by travisghansen** [comment]
@ldez how is this not a bug?  The port is entirely irrelevant to actual behavior and yet is required.

**Comment by ldez** [comment]
Maybe I have misunderstood your issue.

Could you edit your first message and add more information:
- detailed steps to reproduce
- full configuration
- logs

**Comment by travisghansen** [comment]
@ldez I'm running with the `swarm` backend.  `traefik` is launched as such (minor things removed for privacy/security):
```
docker service create \
  --publish 81:80 \
  --name=traefik \
  --constraint=node.role==manager \
  --mount type=bind,source=/var/run/docker.sock,target=/var/run/docker.sock \
  --label traefik.enable=true \
  --label traefik.port=8080 \
  --label traefik.docker.network=services \
  --label traefik.frontend.auth.basic='' \
  --network services \
  traefik:v1.3.2 \
    --docker \
    --docker.swarmmode=true \
    --docker.exposedbydefault=false \
    --docker.watch=true \
    --docker.domain='' \
    --web
```
I'm trying to launch a container exposing multiple ports (services in `traefik` parlance from what I can tell) like such:
```
docker service create \
  --name=kannel \
  --publish 13000:13000 \
  --publish 13013:13013 \
  --mount type=bind,source=/vault/docker/kannel/config,target=/etc/kannel/conf.d \
  --mount type=bind,source=/vault/docker/kannel/ssl,target=/etc/kannel/ssl \
  --label traefik.enable=true \
  --label traefik.port=80 \
  --label traefik.docker.network=services \
  --label traefik.enable=true \
  --label traefik.admin.port=13000 \
  --label traefik.admin.frontend.rule=Host:blah-here \
  --label traefik.admin.frontend.auth.basic='' \
  --label traefik.sendsms.port=13013 \
  --label traefik.sendsms.frontend.rule=Host:blah-there \
  --network services \
  registry/docker-kannel
```
The above works great, however if I simply omit `--label traefik.port=80` then none of the services are registered/detected by traefik.  Also note that, as is, *nothing* registers with port `80` (which is what I want because it's bogus anyhow).

If I do something like this:
```
docker service create \
  --name=kannel \
  --publish 13000:13000 \
  --publish 13013:13013 \
  --mount type=bind,source=/vault/docker/kannel/config,target=/etc/kannel/conf.d \
  --mount type=bind,source=/vault/docker/kannel/ssl,target=/etc/kannel/ssl \
  --label traefik.enable=true \
  --label traefik.docker.network=services \
  --label traefik.enable=true \
  --label traefik.port=13000 \
  --label traefik.frontend.rule=Host:blah-here \
  --label traefik.frontend.auth.basic='' \
  --label traefik.sendsms.port=13013 \
  --label traefik.sendsms.frontend.rule=Host:blah-there \
  --network services \
  registry/docker-kannel
```
Then only the `sendsms` service is registered and not the *default* service running on port `13000`.  Which aligns nicely with the behavior described in the earlier scenario.

So:
 1. If `traefik.port` is not set *nothing* is registered.
 1. If `traefik.port` is set, the *default* (non-namespaced) service **is completely ignored**.  It's a grand conundrum :)

I feel as if a few things need to be improved:
 1. `traefik.port` should not be **required** as long as *any* `traefik.<service>.port` entries exist.
 1. The documentation could be expanded to clarify the nuances above (ie: you can use *either* namespaced services, *or* the default, but not *both*.
 1. Or update the code to allow/detect *both*.  I feel like using one or the other makes sense (as is) particularly if the doc explained the behavior.  However, as originally mentioned, when using namespaced services `traefik.port` shouldn't be **required** to make them work.  It's super confusing from multiple angles.

Thanks for the consideration and damn cool product!

**Comment by bitsofinfo** [comment]
+1 experiencing same

**Comment by bitsofinfo** [comment]
however I get this error when scaling the service beyond one (using latest traefik version)

```
2017-07-11T13:55:03.921062141Z time="2017-07-11T13:55:03Z" level=error msg="Near line 27 (last key parsed 'backends.backend-xxx-docker-swarm-restlet.servers'): Key 'backends.backend-xxx-docker-swarm-restlet.servers.service' has already been defined."
```

**Comment by travisghansen** [comment]
@bitsofinfo can you send over a sample series of commands for me to replicate?  I'm currently using this is a small setup and haven't messed with the scale-out features beyond simply using `mode: global`.

**Comment by benoitm76** [comment]
\+ 1 I am experiencing also this issue on my swarm cluster.

I found a work around to make it works :

```yaml
labels:
    traefik.port: "8081"
    traefik.frontend.rule: "Host: nexus.example.com"
    traefik.docker.network: "traefik_network"
    traefik.nexus.port: "8081"
    traefik.nexus.frontend.rule: "Host: nexus.example.com"
    traefik.nexus.docker.network: "traefik_network"
    traefik.registry.port: "6253"
    traefik.registry.frontend.rule: "Host: registry.example.com"
    traefik.registry.docker.network: "traefik_network"
```

If I not put `traefik.port`, `traefik.frontend.rule` and `traefik.docker.network` my nexus container is not registered.

I can't also remove `traefik.nexus.port`, `traefik.nexus.frontend.rule` and `traefik.nexus.docker.network` because in this case nexus service is ignored (`traefik.port`, `traefik.frontend.rule` and `traefik.docker.network` are necessary but not take in consideration).

After a quick look, the problem can comes from this part of the code I think: `traefik/provider/docker/docker.go`
```go
func (p *Provider) getFrontendRule(container dockerData) string {
	if label, err := getLabel(container, types.LabelFrontendRule); err == nil {
		return label
	}
	if labels, err := getLabels(container, []string{labelDockerComposeProject, labelDockerComposeService}); err == nil {
		return "Host:" + p.getSubDomain(labels[labelDockerComposeService]+"."+labels[labelDockerComposeProject]) + "." + p.Domain
	}
	if len(p.Domain) > 0 {
		return "Host:" + p.getSubDomain(container.ServiceName) + "." + p.Domain
	}
	return ""
}
```
This function is called by `containerFilter` in the same file.

Indeed, the filter will search for a container with a label : `traefik.frontend.rule`.

**Comment by bitsofinfo** [comment]
@travisghansen see this ticket for what I am doing (example) https://github.com/containous/traefik/issues/1853

**Comment by traefiker** [comment]
Closed by #2330.
